plugins/woocommerce-admin/client/products/fields/category-field/create-category-modal.tsx

```diff
@@ -0,0 +1,192 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, Modal, Spinner, TextControl } from '@wordpress/components';
+import { useMemo, useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import {
+	__experimentalSelectControl as SelectControl,
+	__experimentalSelectControlMenu as Menu,
+} from '@woocommerce/components';
+import { recordEvent } from '@woocommerce/tracks';
+import {
+	EXPERIMENTAL_PRODUCT_CATEGORIES_STORE_NAME,
+	ProductCategory,
+} from '@woocommerce/data';
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import './create-category-modal.scss';
+import { useCategorySearch } from './use-category-search';
+import { CategoryFieldItem } from './category-field-item';
+
+type CreateCategoryModalProps = {
+	initialCategoryName?: string;
+	onCancel: () => void;
+	onCreated: ( newCategory: ProductCategory ) => void;
+};
+
+export const CreateCategoryModal: React.FC< CreateCategoryModalProps > = ( {
+	initialCategoryName,
+	onCancel,
+	onCreated,
+} ) => {
+	const {
+		categoriesSelectList,
+		isSearching,
+		categoryTreeKeyValues,
+		searchCategories,
+		getFilteredItems,
+	} = useCategorySearch();
+	const { createNotice } = useDispatch( 'core/notices' );
+	const [ isCreating, setIsCreating ] = useState( false );
+	const { createProductCategory, invalidateResolutionForStoreSelector } =
+		useDispatch( EXPERIMENTAL_PRODUCT_CATEGORIES_STORE_NAME );
+	const [ categoryName, setCategoryName ] = useState(
+		initialCategoryName || ''
+	);
+	const [ categoryParent, setCategoryParent ] = useState< Pick<
+		ProductCategory,
+		'id' | 'name'
+	> | null >( null );
+
+	const onSave = async () => {
+		recordEvent( 'product_category_add', {
+			new_product_page: true,
+		} );
+		setIsCreating( true );
+		try {
+			const newCategory: ProductCategory = await createProductCategory( {
+				name: categoryName,
+				parent: categoryParent ? categoryParent.id : undefined,
+			} );
+			invalidateResolutionForStoreSelector( 'getProductCategories' );
+			setIsCreating( false );
+			onCreated( newCategory );
+		} catch ( e ) {
+			createNotice(
+				'error',
+				__( 'Failed to create category.', 'woocommerce' )
```

Comment:
I know error handling wasn't the specific focus of this PR, but long-term we need to come up with better error handling around more common errors across all product components.

Duplicate slugs are a common one that I've been hitting on accident when testing and we need more specific error messages to help users navigate these issues.